### PR TITLE
Add CI/CD in Github Actions

### LINKED ISSUE #440: move all ci/cd to github actions, including testing, linting, and codecov
our `.github/workflows` only contains an action for testing, it would be good to add the linting step in the Makefile and also add a step to check codecov

### PR DESCRIPTION
**Pull request by rodrigoalmeida94**
- Fix linter errors
- Adds CI/CD for testing
- Adds CI/CD for release

### CODE REVIEW

**Comment by rodrigoalmeida94** [comment]
@srmsoumya you can close #446 in favour of this PR. Let me know if this looks good!

File: tox.ini [hunk 1]
```diff
@@ -60,3 +47,31 @@ deps =
...
+max-complexity = 10
+count=true
+
+[isort]
+profile = black
+
+[black]
+line-length = 88
+target-version = ['py37', 'py38', 'py39', 'py310']
+experimental_string_processing = true
```
Comment (vincentsarago) [inline tox.ini]: @rodrigoalmeida94 what do you think about moving those configurations (flake, isort, black) into a `setup.cfg` file. While having it in `tox.ini` is technically ok, I think `setup.cfg` is more supported.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -38,10 +38,16 @@ jobs:
...
+            libgl1
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install tox tox-gh-actions
+          python -m pip install tox
+      - name: Lint with tox
+        run: tox -e lint
+      - name: Format with tox
+        run: tox -e format
```
Comment (vincentsarago) [inline .github/workflows/tests.yml:50]: is `pre-commit` something that could be added in the repo? then we could just run `pre-commit run --all-files`
https://github.com/developmentseed/titiler/blob/master/.github/workflows/ci.yml#L55-L56

File: tox.ini (hunk 1 above)
Comment (rodrigoalmeida94) [reply tox.ini]: Good idea @vincentsarago !! Added.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (rodrigoalmeida94) [reply .github/workflows/tests.yml:50]: Very good idea, Added a pre-commit config, @vincentsarago would you suggest removing lint and format completely out of the tox file?

Comment (rbavery) [reply .github/workflows/tests.yml:50]: @rodrigoalmeida94 approved! though I would wait to hear what Vincent things about removing lint and format from tox before merging.

**Review by rbavery — APPROVED** [review]

**Review by srmsoumya — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 2 above)
Comment (rodrigoalmeida94) [reply .github/workflows/tests.yml:50]: It's quite a tricky thing it seems to work with different python version in the CI for running tests and using pre-commit, leaving in the tox format and lint as is in github actions as a sanity check.

**Pull request merged**